server/src/main/java/org/opensearch/cluster/metadata/MetadataCreateIndexService.java

```diff
@@ -566,6 +562,22 @@ IndexMetadata buildAndValidateTemporaryIndexMetadata(
         return tempMetadata;
     }
 
+    public void addRemoteCustomData(IndexMetadata.Builder tmpImdBuilder) {
```

Comment:
Assuming custom metadata would be available during restore for snapshots taken on index metadata that had it enabled. This is only for snapshots that might not have indices enabled with custom paths

Reply:
Yes, that's right.